packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,57 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule<[], 'arrayDeleteViolation'>({
+  name: 'no-array-delete',
+  meta: {
+    docs: {
+      description: 'Disallow delete operator for arrays',
+      recommended: 'error',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      arrayDeleteViolation:
```

Comment:
Nit: "violation" is implied by the fact that it's a message.

```suggestion
      arrayDelete:
```